### PR TITLE
doc: add kernel parameters documentation

### PR DESCRIPTION
**Pull request by dbkinder**
Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/developer-guides/GVT-g-kernel-options.rst [hunk 1]
```diff
@@ -67,6 +67,8 @@ Currently, all UOSes share the same priority.
 This is a Service OS only parameters, and does
 not work in the User OS.
 
+.. _i915-enable-inital-modeset:
```
Comment (gvancuts) [inline doc/developer-guides/GVT-g-kernel-options.rst]: Nitpick, could you update this ref to:
```suggestion
.. _i915-enable-initial-modeset:
```

Comment (gvancuts) [inline doc/developer-guides/GVT-g-kernel-options.rst]: Nitpick, can you update this to:
```suggestion
.. _i915-enable-initial-modeset:
```

File: doc/developer-guides/kernel-parameters.rst [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+     - SOS,UOS
+     - Force enable atomic functionality on platforms that don't have full support yet.
+     - ::
+
+         i915.nuclear_pageflip=1
+
+   * - i915.enable_initial_modeset
+     - SOS
+     - On MRB, value must be ``1``.  On NUC or UP2 boards, value must be
+       ``0``. See :ref:`i915-enable-inital-modeset`.
```
Comment (gvancuts) [inline doc/developer-guides/kernel-parameters.rst]:
```suggestion
       ``0``. See :ref:`i915-enable-initial-modeset`.
```

Well, only if you accept my suggestion of course :-)

File: doc/developer-guides/kernel-parameters.rst [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+   * - i915.enable_guc
+     - SOS
+     - Enable GuC load for HuC load.
+     - ::
+
+         i915.enable_guc=0x02
+
+   * - i915.avail_planes_per_pipe
+     - UOS
+     - See :ref:`i915-enable-inital-modeset`.
```
Comment (gvancuts) [inline doc/developer-guides/kernel-parameters.rst]:
```suggestion
     - See :ref:`i915-enable-initial-modeset`.
```
Well, only if you accept my suggestion of course :-)

File: doc/developer-guides/kernel-parameters.rst [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+   :widths: 10,10,50,30
+
+   * - Parameter
+     - Used in SOS or UOS
+     - Description
+     - Usage example
+
+   * - module_blacklist
+     - SOS
+     - A comma-separated list of modules that should not be loaded. Useful for debugging problem modules.
```
Comment (gvancuts) [inline doc/developer-guides/kernel-parameters.rst]: This reads a bit strange to me: "Useful for debugging problem modules." Can we improve it? Maybe something like "Useful to debug or work around issues related to specific modules"

File: doc/developer-guides/kernel-parameters.rst [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+          log_buf_len=16M
+
+   * - consoleblank
+     - SOS,UOS
+     - The console blank (screen saver) timeout in
+       seconds. Defaults to 600 (10 minutes). A value of 0
+       disables the blank timer.
+     - ::
+
+          consoleblack=0
```
Comment (gvancuts) [inline doc/developer-guides/kernel-parameters.rst]:
```suggestion
          consoleblank=0
```

File: doc/developer-guides/kernel-parameters.rst (hunk 3 above)
Comment (gvancuts) [reply doc/developer-guides/kernel-parameters.rst]: Actually, this is the wrong reference. It should be
```suggestion
     - See :ref:`i915-avail-planes-owners`.
```

**Comment by dbkinder** [comment]
@gvancuts suggested changes made

**Pull request merged**